app/story/seo_validations.js

```diff
@@ -0,0 +1,132 @@
+const _ = require("lodash");
+
+const _occurences = (content, keyword) => {
+  let matches = content && content.toLowerCase().match(new RegExp(keyword.toLowerCase()));
+  return (matches && matches.length) || 0;
+};
+
+const _containsKeyword = (content, keyword) => {
+  return content && keyword && _occurences(content, keyword) > 0;
+};
+
+const _wordCount = content => {
+  return content && content.split(/\s+/).filter(word => {
+    return word !== "";
+  }).length;
+};
+
+const _errorType = config => {
+    return config['throws']
+}
+
+const validatePresence = (ruleConfig, output, data) => {
+  if (!data.content || data.content.length === 0) {
+    output[_errorType(ruleConfig)].push(`Empty ${data.field}`);
+  } else if (data.content.length > 0) {
+    output.goodies.push(`You've added ${data.field}`);
+  }
+};
+
+const validateFocus = (ruleConfig, output, data) => {
+  if (!data.focusKeyword) return;
+  if (data.content && !_containsKeyword(data.content, data.focusKeyword)) {
+    output[_errorType(ruleConfig)].push(
+      `The ${data.field} doesn't contain focus keyword '${data.focusKeyword}'.`
+    );
+  } else {
+    output.goodies.push(
+      `The ${data.field} contains focus keyword '${data.focusKeyword}'.`
+    );
+  }
+};
+
+const validateImageAltText = (ruleConfig, output, data) => {
+  let images = data.content;
+  if (images && _.some(images, image => !image["alt-text"])) {
+    output[_errorType(ruleConfig)].push("Few images don't contain Alt attribute");
+  } else {
+    output.goodies.push("You have used Alt attribute in all images");
+  }
+};
+
+const validateImageAltTextFocus = (ruleConfig, output, data) => {
+  let focusKeyword = data.focusKeyword;
+  if (!focusKeyword) return;
+  let images = data.content;
+  let altTexts = images.map(image => image["alt-text"]);
+  if (_.some(altTexts, altText => _containsKeyword(altText, focusKeyword))) {
+    output.goodies.push(
+      `You\'ve used the focus keyword '${focusKeyword}' in the alt tag of an image`
+    );
+  } else {
+    output[_errorType(ruleConfig)].push(
+      `You have not used focus keyword '${focusKeyword}' in the alt tag of image`
+    );
+  }
+};
+
+const validateLetterCount = (ruleConfig, output, data) => {
+  if (data.content && data.content.length > 0 && data.content.length < ruleConfig.min) {
+    output[_errorType(ruleConfig)].push(`The ${data.field} is too short.`);
```

Comment:
In all the messages with data being short or long it would be better if we can add the info about what the min and max values of what the data length should be.

Reply:
This is part of prev rules. Will handle this after merging this